src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Contracts/NonTSqlNotification.cs

```diff
@@ -0,0 +1,741 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#nullable disable
+
+using System.Collections.Generic;
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.LanguageServices.Contracts
+{
+    /// <summary>
+    /// Parameters to be sent back with a non-TSQL file event.
+    /// </summary>
+    public class NonTSqlParams
+    {
+        /// <summary>
+        /// URI identifying the file detected as non-TSQL.
+        /// </summary>
+        public string OwnerUri { get; set; }
+
+        /// <summary>
+        /// Indicates whether the file was flagged due to containing
+        /// non-TSQL keywords or hitting the error limit.
+        /// </summary>
+        public bool ContainsNonTSqlKeywords { get; set; }
+    }
+
+    /// <summary>
+    /// NonTSqlNotification notification mapping entry.
+    /// </summary>
+    public class NonTSqlNotification
+    {
+        public static readonly EventType<NonTSqlParams> Type =
+            EventType<NonTSqlParams>.Create("textDocument/nonTSqlFileDetected");
+    }
+
+    public static class TSqlDetectionConstants
+    {
+        /// <summary>
+        /// The error threshold for when a file is considered non-TSQL.
+        /// </summary>
+        /// 
+        public const int SqlFileErrorLimit = 50;
+
+        /// <summary>
+        /// A list of keywords that exist in MySQL and PostgreSQL documentation
+        /// but are not present in T-SQL documentation.
+        /// Sources:
+        /// - MySQL: https://dev.mysql.com/doc/refman/8.4/en/keywords.html
+        /// - PostgreSQL: https://www.postgresql.org/docs/current/sql-keywords-appendix.html
+        /// - T-SQL: https://learn.microsoft.com/en-us/sql/t-sql/language-elements/reserved-keywords-transact-sql
+        /// </summary>
+        public static readonly HashSet<string> Keywords = new HashSet<string>
+        {
+            "COPARTITION",
```

Comment:
Can you make this list in alphabetical order? vscode should have an extension that will let you sort lines alphabetically.

If possible, can you separate this list into Postgres/MYSQL, etc?